### PR TITLE
add some instructions for HDF5-dependency

### PR DESCRIPTION
**Pull request by hollandjg**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,8 @@
 Prerequisites:
 - A POSIX-compatible system (Linux or macOS)
 - Python 3.9 (run `python --version` to check that your version of python is correct)
-- MPI (e.g. from `brew install open-mpi`)
+- MPI (e.g. from `brew install open-mpi` on macOS)
+- HDF5 (e.g. from `brew install hdf5` on macOS)
```
Comment (cpaniaguam) [inline README.md:9]: Is this a _true_ prerequisite? I tested without having to explicitly install hdf5 and it worked fine.

Comment (kmilo9999) [reply README.md:9]: I believe it is necessary. Maybe you already had hdf5 installed from a previous project or library.

**Review by kmilo9999 — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -31,6 +32,18 @@ Installation:
   ```shell
   pip install --upgrade --editable ".[dev]"
   ```
+  > You may need to set the value of the `HDF5_DIR` environment variable to install some of the dependencies, especially when installing on macOS. For Apple Silicon (M-Series) CPUs:
```
Comment (cpaniaguam) [inline README.md:35]:
```suggestion
  > You may need to set the value of the `HDF5_DIR` environment variable to install some of the dependencies, especially when installing on macOS.
  
For Apple Silicon (M-Series) CPUs:
```

**Review by cpaniaguam — CHANGES_REQUESTED** [review]
Suggesting adding a line for consistency. Thanks!

**Review by cpaniaguam — APPROVED** [review]
🚀

**Pull request merged**